### PR TITLE
Fix disable third party requests

### PR DESCRIPTION
**Pull request by saghul**
Fixes: https://github.com/jitsi/jitsi-meet/issues/5446

### CODE REVIEW

File: react/features/base/lib-jitsi-meet/functions.any.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function createLocalTrack(type: string, deviceId: string) {
 export function isAnalyticsEnabled(stateful: Function | Object) {
     const { disableThirdPartyRequests, analytics = {} } = toState(stateful)['features/base/config'];
 
-    return !disableThirdPartyRequests && !analytics.disabled;
+    return !disableThirdPartyRequests || !analytics.disabled;
```
Comment (bgrozev) [inline react/features/base/lib-jitsi-meet/functions.any.js]: This doesn't look right. Did you want to change it to `!(disableThirdPartyRequests || analytics.disabled)`?

Comment (saghul) [reply react/features/base/lib-jitsi-meet/functions.any.js]: 🤦‍♂

**Comment by saghul** [comment]
Great catch @bgrozev ! Updated.

**Comment by rosa2** [comment]
@saghul thanks a lot for your good work. Sorry to write in a closed issue.
Why if I go to
https://meet.jit.si/foo#config.disableThirdPartyRequests=true

Google Analytics is there still? Is it because the server has not updated? I check with https://chrome.google.com/webstore/detail/noscript/doojmbjmlfjjnbmnoijecmcbfeoakpjm

Thanks again

**Comment by saghul** [comment]
Hey @rosa2 ! This change has not yet been deployed to meet.jit.si, it should happen in the upcoming days.

**Comment by rosa2** [comment]
Gràcies/Gracias/Thanks @saghul 
I hope it is soon because in https://fediverse.blog/~/DonsBlog/videochat-server they have you as a bad choice. I wait for it to work to spread the new :)

**Comment by saghul** [comment]
They can put us wherever the autors of that list feel like. We will keep on doing the good work I think we're doing.

**Comment by Jip-Hop** [comment]
I just tried to see if the fix is already live at beta.meet.jit.si. It's using version [3940](https://github.com/jitsi/jitsi-meet/releases/3940) which is from 9 hours ago, so it should have this fix.

And I can confirm it's working as longs as you make sure you put the config override in the url: https://beta.meet.jit.si/foo#config.disableThirdPartyRequests=true. It finishes loading after having made approximately 30 requests according to the Network tab in Chrome Dev Tools. Google Analytics not among them. Great 😄

The hash value disappears once the page is loaded, so don't forget to add it. If you make a webpage (e.g. GitHub Pages) and use the external API you could permanently 'host' your config overrides while still using the official meet.jit.si instance (or the beta one for latest changes). That way others can easily use the config override values you've setup. For example:

```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>Jitsi Meet - No Tracking</title>
    <script src="https://beta.meet.jit.si/external_api.js"></script>
    <style>
      * {
        margin: 0;
        padding: 0;
        border: 0;
      }

      html,
      body {
        width: 100vw;
        height: 100vh;
      }
    </style>
  </head>

  <body>
    <script>
      const domain = "beta.meet.jit.si";
      const options = {
        roomName: "disableThirdPartyRequestsTest",
        parentNode: document.body,
        configOverwrite: { disableThirdPartyRequests: true }
      };
      const api = new JitsiMeetExternalAPI(domain, options);
    </script>
  </body>
</html>
```
Unfortunately you MUST specify a room name in this case. If you don't, you'll get the default Jitsi entry page where you can pick a room name etc. (which is nice) but when you hit "GO" it will still load Google Analytics, meaning `configOverwrite: { disableThirdPartyRequests: true }` is gone.

So if you go this route, you'll need to make your own entry page (a simple login form would do) and init Jitsi on submit via the External API with the chosen room name.

I've done exactly that and a live example is [here](https://madtech.me/jitsi/). Server 1 is beta.meet.jitsi. But none of the servers in the dropdown load Google Analytics, since I also used `config.analytics.disabled=true`, which was not broken.

@saghul yes, please keep on doing the good work 🙂 thanks!

**Comment by saghul** [comment]
Thanks @Jip-Hop ! Please don't link beta publicly as it's not meant for public beta testing.

**Comment by Jip-Hop** [comment]
Sorry, wasn't aware of that. Does that instance have less resources?

I've removed beta.meet.jit.si from my login form.

**Comment by saghul** [comment]
No worries! Yes, it has less resoures. We occasionally ask the community to validate fixes there, but it's not meant to be a daily driver for the public.

**Pull request merged**